### DISCUSSION TITLE
how to retrieve workflow information by api?

### DISCUSSION THREAD

**Discussion by rezabashiri** [elsa-workflows/elsa-core#474] [Q&A]
I need to know current state of a specific workflow by api . is there any way to pass instanceId to a api to fetch workflow status ?

**Comment by sfmskywalker** [comment]
Yes, in Elsa 1 you would use the `IWorkflowInstanceStore` service and its `GetByIdAsync` method to get a single workflow instance by ID.

**Reply by rezabashiri** [reply]
is that possible to filter by activity property? I mean find all instances which have particular value in a particular property?

**Reply by sfmskywalker** [reply]
Not yet, but this will be possible once we have Lucene Indexing implemented. Then any activity type can provide its own indexing handler, allowing its properties to be made part of the index.

**Reply by rezabashiri** [reply]
I'm trying to implement a cardboard system that each user can access to his own jobs being passed to her/him, so filtering instance based on users' role is what I need, what is your solution to this challenge?

**Reply by sfmskywalker** [reply]
One thing you could do right now with Elsa 2 is implement a `WorkflowSuspended` event handler. Assuming you associated the workflow with a user, you can access this user information and create / update / delete records from a custom DB table where you associate the user ID and/or user role with the workflow instance ID. Your application can then simply perform a lookup in that table by user ID or user role.
